prow/external-plugins/cherrypicker/server.go

```diff
@@ -168,7 +167,7 @@ func (s *Server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent
 	commentAuthor := ic.Comment.User.Login
 
 	// Do not create a new logger, its fields are re-used by the caller in case of errors
-	l = l.WithFields(logrus.Fields{
+	*l = *l.WithFields(logrus.Fields{
```

Comment:
This didn't do what the comment above claimed, `l` holds a pointer, the assignment before made it hold a new pointer, but that won't change the original pointer. Changing it like this will instead change the value behind the pointer, so the caller will actually get these fields as well.